Additional restructuring suggestions for CAP Java Remote Services section (#796)
cap-js/docs: java/cqn-services/remote-services.md

These service instances of BTP services provide the URL of the remote API in their service binding.
Therefore you only need to specify the binding name in the `application.yaml` configuration, like in the following example:

```yaml
cds:
  remote.services:
    SomeReuseService:
      binding:
        name: some-service-binding
```

In some cases, SAP Cloud SDK doesn't understand the service binding structure of the specific BTP service.
In that case it's required to contribute a mapping by the means of Cloud SDK's `PropertySupplier`.
This `PropertySupplier` needs to be registered with the Cloud SDK once at application startup.

```java
static {
    OAuth2ServiceBindingDestinationLoader.registerPropertySupplier(
        options -> options.getServiceBinding().getTags().contains("<bindingTag>"),
            SomeReuseServiceOAuth2PropertySupplier::new);
}
```

The `<bindingTag>` needs to be replaced by the concrete name of the tag provided in the binding of the BTP service. Alternatively, a check on the service name can be chosen as well.
The class `SomeReuseServiceOAuth2PropertySupplier` needs to be provided by you extending the Cloud SDK base class `DefaultOAuth2PropertySupplier`.

> [Learn more about registering OAuth2PropertySupplier in the **SAP Cloud SDK documentation**.](https://sap.github.io/cloud-sdk/docs/java/features/connectivity/service-bindings#customization){.learn-more}

Reviewer: Can we put that in a details section? Seems to be an edge case/troubleshooting, right?

Reviewer: Expanded:
![image](https://github.com/cap-js/docs/assets/32062331/4b43cc28-6a24-4ccc-9a73-d277552c0b0e)

Collapsed:
![image](https://github.com/cap-js/docs/assets/32062331/0f2b21f1-9b03-45df-b164-54e2c674ebac)

Author: +1

#### Binding to a Service with shared Identity

> In case the remote API is available within the same SaaS application and using the same (shared) XSUAA service instance for authentication no service broker-based reuse service is required.

Reviewer: ```suggestion
If the remote API is available within the same SaaS application and using the same (shared) XSUAA service instance for authentication, no service broker-based reuse service is required.
```

```java
ServiceBinding binding = ...;
HttpDestination destination = ServiceBindingDestinationLoader.defaultLoaderChain().getDestination(
        ServiceBindingDestinationOptions
          .forService(binding)
          .onBehalfOf(OnBehalfOf.TECHNICAL_USER_CURRENT_TENANT)
          .build());

HttpClient httpClient = HttpClientAccessor.getHttpClient(destination);
...
```

[Learn more about HttpClientAccessor in the **SAP Cloud SDK documentation**.](https://sap.github.io/cloud-sdk/docs/java/features/connectivity/http-client){.learn-more}

To be able to resolve a service binding into a Cloud SDK destination, a `OAuth2PropertySupplier` might need to be registered with Cloud SDK.

```java
static {
    OAuth2ServiceBindingDestinationLoader.registerPropertySupplier(
        options -> options.getServiceBinding().getTags().contains("<bindingTag>"),
            BizPartnerOAuth2PropertySupplier::new);
}
```

[Learn more about registering OAuth2PropertySupplier in the **SAP Cloud SDK documentation**.](https://sap.github.io/cloud-sdk/docs/java/features/connectivity/service-bindings#customization){.learn-more}

### Using Destinations { #native-destinations }

> If the URL and credentials of the remote API is configured are available as a destination in SAP BTP Destination Service, you can use Cloud SDK's `DestinationAccessor` API to load the destination based on its name. In a second step, `HttpClientAccessor` is used to create an instance of `HttpClient`:

Reviewer 2: ```suggestion
If the URL and credentials of the remote API is configured as a destination in SAP BTP Destination Service, you can use Cloud SDK's `DestinationAccessor` API to load the destination based on its name. In a second step, `HttpClientAccessor` is used to create an instance of `HttpClient`:
```
